src/mqt/bench/benchmark_generation.py

```diff
@@ -300,225 +153,217 @@ def get_native_gates_level(
-        "qftentangled",
-        "qnn",
-        "qpeexact",
-        "qpeinexact",
-        "qwalk",
-        "randomcircuit",
-        "vqerealamprandom",
-        "vqesu2random",
-        "vqetwolocalrandom",
-        "wstate",
-        "shor",
-    ]
+def get_mapped_benchmark(
+    benchmark: str | QuantumCircuit,
+    circuit_size: int | None = None,
+    benchmark_instance_name: str | None = None,
+    opt_level: int | None = 2,
+    target: Target = None,
+) -> QuantumCircuit:
+    """Return a benchmark fully compiled and qubit-mapped to a device.
 
+    Arguments:
+            benchmark: QuantumCircuit or name of the benchmark to be generated
+            circuit_size: Input for the benchmark creation, in most cases this is equal to the qubit number
+            benchmark_instance_name: Input selection for some benchmarks, namely "shor"
+            opt_level: Optimization level to be used by the transpiler.
+            target: `~qiskit.transpiler.target.Target` for the benchmark generation
 
-def get_supported_levels() -> list[str | int]:
-    """Returns a list of all supported benchmark levels."""
-    return ["alg", "indep", "nativegates", "mapped", 0, 1, 2, 3]
+    Returns:
+            Qiskit::QuantumCircuit that has been decomposed and routed onto the connectivity described by ``target``.
+    """
+    if target is None:
+        msg = "`target` must be supplied for the mapped level."
+        raise ValueError(msg)
 
+    opt_level = _normalise_opt_level(opt_level)
 
-def get_module_for_benchmark(benchmark_name: str) -> ModuleType:
-    """Returns the module for a specific benchmark."""
-    return import_module("mqt.bench.benchmarks." + benchmark_name)
+    qc = _create_raw_circuit(benchmark, circuit_size, benchmark_instance_name)
+    return _to_mapped(qc, device=target, opt_level=opt_level)
+
+
+def get_benchmark(
+    benchmark: str | QuantumCircuit,
+    level: str | int,
+    circuit_size: int | None = None,
+    benchmark_instance_name: str | None = None,
+    opt_level: int | None = 2,
+    target: Target | None = None,
+) -> QuantumCircuit:
+    """Returns one benchmark as a qiskit.QuantumCircuit object.
+
+    Arguments:
+        benchmark: QuantumCircuit or name of the benchmark to be generated
+        level: Choice of level, either as a string ("alg", "indep", "nativegates" or "mapped") or as a number between 0-3 where 0 corresponds to "alg" level and 3 to "mapped" level
+        circuit_size: Input for the benchmark creation, in most cases this is equal to the qubit number
+        benchmark_instance_name: Input selection for some benchmarks, namely "shor"
+        opt_level: Optimization level to be used by the transpiler.
+        target: `~qiskit.transpiler.target.Target` for the benchmark generation (only used for "nativegates" and "mapped" level)
+
+    Returns:
+        Qiskit::QuantumCircuit object representing the benchmark with the selected options
+    """
+    level_map: dict[str | int, str] = {
```

Comment:
I think we should actually move away from having both the `str` *and* the `int` description of the level. I would prefer to only have the `str` one, using an enum would probably even nicer. This should reduce the complexity and different possible cases to make the code easier.